doc/api/async_hooks.md

```diff
@@ -712,6 +712,75 @@ never be called.
 * Returns: {number} The same `triggerAsyncId` that is passed to the
 `AsyncResource` constructor.
 
+## Class: AsyncContext
+
+This class is used to create contexts that can be used through the event loop.
+
+### Example
+
+Let's build a logger that will always know the current HTTP request and use
+it to display enhanced logs without needing to explicitly pass the current
+HTTP request to it.
+
+```js
+const { AsyncContext } = require('async_storage');
+const http = require('http');
+
+const kReq = 'CURRENT_REQUEST';
+const asyncContext = new AsyncContext();
+
+function log(...args) {
+  const store = asyncContext.getStore();
+  // Let's make sure the store exists and it contains a request
+  if (store && store.has(kReq)) {
+    const req = store.get(kReq);
+    // For instance, prints `GET /items ERR could not do something
+    console.log(req.method, req.url, ...args);
+  } else {
+    console.log(...args);
+  }
+}
+
+http.createServer((request, response) => {
+  const store = asyncContext.enter();
+  store.set(kReq, request);
+  // some code
+   someAsyncOperation((err, result) => {
+    if (err){
+      log('ERR', err.message);
+      // ... rest of the code
+    }
+  });
+})
+.listen(8080);
+```
+
+### new AsyncContext()
+
+Creates a new instance of AsyncContext. Until the `enter` method is called, it
+does not provide any storage features.
+
+### asyncContext.enter()
+
+Calling `asyncContext.enter()` will create a new asynchronous context.
```

Comment:
One different here from what I see in some other cls implementations (for example 'continuation-local-storage') is that this seems to force you to pass around the store returned as opposed to creating it with a namespace that can later be used to retrieve it.

Reply:
another pattern to consider is zones

https://github.com/domenic/zones
https://gist.github.com/devsnek/4128215236f40dc7939b269a50db5646

Reply:
With the name you can do something like `const parentSpan = cls.getNamespace('http').get('span');` as opposed to having access to  the AsyncContext that was created earlier.  On the other hand with `continuation-local-storage`  there was extra of binding a namespace object to functions (or running function with the run command) .